File: Ser2025/Company/Company.Web/Modules/ServerTypes/Company/CurrencyColumns.ts
import { ColumnsBase, fieldsProxy } from '@serenity-is/corelib';
import { Column } from '@serenity-is/sleekgrid';
import { CurrencyRow } from './CurrencyRow';

export interface CurrencyColumns {
    CurrencyCode: Column<CurrencyRow>;
    Name: Column<CurrencyRow>;
    IsActive: Column<CurrencyRow>;
    TenantId: Column<CurrencyRow>;
}

export class CurrencyColumns extends ColumnsBase<CurrencyRow> {
    static readonly columnsKey = 'Company.Currency';
    static readonly Fields = fieldsProxy<CurrencyColumns>();
}

File: Ser2025/Company/Company.Web/Modules/Company/Currency/CurrencyColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace Company.Company.Columns;

[ColumnsScript("Company.Currency")]
[BasedOnRow(typeof(CurrencyRow), CheckNames = true)]
public class CurrencyColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public string CurrencyCode { get; set; }
    public string Name { get; set; }
}